docs/docs/policies.mdx

```diff
@@ -396,6 +398,308 @@ for details.
+  This parameter sets the number of times the algorithm will see the training data (default: `1`).
+  One `epoch` is equals to one forward pass and one backward pass of all the training examples.
+  Sometimes the model needs more epochs to learn properly.
+  Sometimes more epochs don't influence the performance.
+  The lower the number of epochs the faster the model is trained.
+  Here is how the config would look like:
+
+  ```yaml-rasa title="config.yml"
+  policies:
+  - name: IntentTEDPolicy
+    epochs: 200
+  ```
+
+* `max_history`:
+  This parameter controls how much dialogue history the model looks at before making an inference.
+  Default `max_history` for this policy is `None`, which means that the complete dialogue history
+  since session restart is taken into account. If you want to limit the model
+  to only see a certain number of previous
+  dialogue turns, you can set `max_history` to a finite value.
+  Please note that you should pick `max_history` carefully, so that the model has enough
+  previous dialogue turns to create a correct prediction.
+  Higher values of `max_history` can result in more frequent prediction of `action_unlikely_intent`
+  as the number of unique possible conversation paths increases as more dialogue context is taken
+  into account. Lowering the value of `max_history` will result in `action_unlikely_intent` being
+  triggered less often but can be a stronger indicator that the corresponding conversation path
+  is highly unique and hence unexpected.
+  Here is how the config would look like:
+
+  ```yaml-rasa title="config.yml"
+  policies:
+  - name: IntentTEDPolicy
+    max_history: 8
+  ```
+
+* `ignore_intents_list`:
+    This parameter lets you configure `IntentTEDPolicy` to ignore the prediction of `action_unlikely_intent` for
+    a subset of intents. You might want to do this if you come across a certain list of intents for which there
+    are too many false warnings generated.
+
+* `tolerance`:
+    The `tolerance` parameter is a number that ranges from `0.0` to `1.0` (inclusive).
+    It helps to adjust the threshold score used during prediction of `action_unlikely_intent`
+    at inference time. Here, `0.0` means that the score threshold is the one
+    that `IntentTEDPolicy` had determined at training time.
+    A tolerance of `1.0` means that the threshold score is so low that `IntentTED` would not
+    trigger `action_unlikely_intent` for any of the "negative examples" that it has encountered
+    during training. These negative examples are combinations of dialogue contexts and user
+    intents that are _incorrect_. `IntentTEDPolicy` generates these negative examples from your
+    training data by picking a random story part and pairing it with a random intent that doesn't
+     occur at this point.
+
+The above configuration parameters are the ones you should try tweaking according to your use case and training data.
+However, additional parameters exist that can be adapted.
```

Comment:
I'd avoid passive voice. So "that you could adapt".

Reply:
I do not understand the explanation of `negative examples`, but at the same time, I wonder if it is needed to explain this here. This might be better covered somewhere else where the internal workings are explained.

Reply:
@ArjaanBuijk Let me try to explain `negative examples` to you and then maybe you could help us to articulate this easier.

If you had one just training story like:

```
story: greet unhappy
steps:
  - intent: greet
  - action: utter_greet
  - intent: mood_happy
  - action: goodbye
```
and you had an `affirm` intent, then a negative example will be:

input: `greet` -> `utter_greet`
output: `affirm`

This is called a negative example because acc to the training stories, `affirm` was never seen as the next intent after `greet` -> `utter_greet` conversation context.

We use these negative examples to figure out the range of values the model predicts for these negative examples constructed for each intent and `tolerance` is a way to be more / less cautious in prediction of `action_unlikely_intent`.

Reply:
@dakshvar22 ,
I re-read the description after seeing your example, and now it is now mostly clear. The one item that is not fully clear is the `score threshold of IntentTEDPolicy`.  Is that explainable in a few words?

A shot at an explanation of the `tolerance` that I would understand is as follows:

---
The `tolerance` parameter is a number that ranges from `0.0` to `1.0` (inclusive).
It helps to adjust the `threshold score` used during prediction of `action_unlikely_intent` at inference time.

During training, `IntentTED` generates "negative examples". For example, if you had one just training story like:

story: greet unhappy
steps:
  - intent: greet
  - action: utter_greet
  - intent: mood_happy
  - action: goodbye
and you had an affirm intent, then a negative example that `IntentTEDPolicy` will generate is:

input: greet -> utter_greet
output: affirm

This is called a negative example because in the training data, affirm was never seen as the next intent after greet -> utter_greet.

For each intent, `IntentTEDPolicy` uses these negative examples to figure out the range of values the model predicts for these negative examples,  and the `tolerance` is a way to be more / less cautious in prediction of `action_unlikely_intent`.
*==> Can we insert a brief explanation of the score threshold here ?*


- A tolerance of `0.0` means that the score threshold (1) is the one that `IntentTEDPolicy` had determined at training time.
- A tolerance of `1.0` means that the threshold score (1,2) is so low that `IntentTED` would not trigger `action_unlikely_intent` for any of the "negative examples" that it has encountered during training. 
---
(1) Is there a difference between `score threshold` and `threshold score` ?
(2) Still struggling to understand what happens when the tolerance is `1.0`. Probably because I do not understand the `score threshold`.

Reply:
@ArjaanBuijk 
> Is there a difference between score threshold and threshold score

No, both are same.

> Still struggling to understand what happens when the tolerance is 1.0. Probably because I do not understand the score threshold.

The threshold score lies between the minimum and maximum scores for all negative examples (for each intent). The exact score is decided on the basis of `tolerance` parameter. If `tolerance` is `0.25` then it means that the threshold score will be selected in such a way that 25% of negative examples wouldn't result in an `action_unlikely_intent`.

Does that help clarify?

Reply:
I have explained this in more detail in "Prediction of `action_unlikely_intent`"  section.